[Fix #118] Fix a false positive for `DeletePrefix` and `DeletePrefix` cops

koic opened the pull request:
Fixes #118.

Fix a false positive for `Performance/DeletePrefix`, `Performance/DeletePrefix`, `Performance/StartWith`, and `Performance/EndWith` cops when receiver is multiline string.

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](https://github.com/rubocop-hq/rubocop-performance/blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop-performance/blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

avdv commented:
Hi @koic 

I think the problem is not in any way related to heredocs.

Take any string containing newlines, and you cannot simply replace a call to `.gsub(/^prefix/, '')` with `delete_prefix('prefix')` since the latter has a different meaning, likewise for `.gsub(/suffix$/`, '')`.

The only thing that works in general is `/\Aprefix/` and `/suffix\z/`.

koic commented:
@avdv Thanks for your feedback. I have rewritten this PR and added `SafeMultiline` option which is `true` by default. So it will detect only `/\Aprefix/` and `/suffix\z/` by default.

avdv commented:
Thank you, LGTM!
